### ISSUE TITLE
i couldnt understand where the initial and boundary condtions  have been used 

### ISSUE THREAD

**Issue by ehtisham409** [maziarraissi/PINNs#8]

**Comment by saiprabhat** [comment]
since the data present in `burgers_shock.mat` was generated using MATLAB, the initial and boundary condition are used to generate this data with the help of a suitable numerical method in MATLAB.

**Comment by sajed-zarrinpour** [comment]
If you mean, where they are used in the neural network?
As far as I could make it by looking at the code, it uses the data implicitly to put bindings on the network.
Here is an example of it in the Schrodinger case:
 In continuous time inference case, in the main function, we have the lower bound and upper bound of the region defined as:
`lb = np.array([-5.0, 0.0])`
`ub = np.array([5.0, np.pi/2])`
and we have
`idx_t = np.random.choice(t.shape[0], N_b, replace=False)`
`tb = t[idx_t,:]`
which is the sampling from those bounds. Moreover, the initial condition can be traced in
`idx_x = np.random.choice(x.shape[0], N0, replace=False)`
`x0 = x[idx_x,:]`
`u0 = Exact_u[idx_x,0:1]`
` v0 = Exact_v[idx_x,0:1]`
Now, if we look at the loss function, we can see this :
`self.loss = tf.reduce_mean(tf.square(self.u0_tf - self.u0_pred)) + \`
 `                 tf.reduce_mean(tf.square(self.v0_tf - self.v0_pred)) + \`
`                    tf.reduce_mean(tf.square(self.u_lb_pred - self.u_ub_pred)) + \`
`                    tf.reduce_mean(tf.square(self.v_lb_pred - self.v_ub_pred)) + \`
 `                   tf.reduce_mean(tf.square(self.u_x_lb_pred - self.u_x_ub_pred)) + \`
`                    tf.reduce_mean(tf.square(self.v_x_lb_pred - self.v_x_ub_pred)) + \`
 `                   tf.reduce_mean(tf.square(self.f_u_pred)) + \`
`                    tf.reduce_mean(tf.square(self.f_v_pred))`
which is indicating that these samples from the lower bound and upper bound and the initial condition are also cooperating in the learning process of the network.
What happens is that the network tries to learn the data you provided for it, which also includes the data over the boundaries. Like so, the network will be forced to respect those conditions in some sense (of course, it treats them just like other data points and it would not be a strong condition since it has the same weight as the other losses. But it has its own impact. I believe one may try to associates some weights to those loss functions to adjust the importance of each of them.).

I would be happy that the community corrects me if I am wrong about it.
